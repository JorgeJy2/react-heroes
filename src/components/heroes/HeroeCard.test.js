import { shallow } from "enzyme";
import { HeroeCard } from "./HeroeCard";

describe('test in card',()=>{

    const mokHeroe = {
        id: 1,
        superhero: 'Name',
        publisher: 'Public',
        alter_ego: 'None',
        first_appearance: 'Today',
        characters: 'fly'
    }

    const wrapper = shallow(<HeroeCard {...mokHeroe}/>);

    test('should snaptshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should data in card', ()=>{
        expect(wrapper.find('img').prop('src')).toBe(`./assets/heroes/${mokHeroe.id}.jpg`);
        expect(wrapper.find('h5').text().trim()).toBe(mokHeroe.superhero);
        expect(wrapper.find('p').at(0).text().trim()).toBe(mokHeroe.alter_ego);
        // if(alter_ego !== characters) {
        //     expect(wrapper.find('p').at(1).text().trim()).toBe(mokHeroe.characters);    
        //     expect(wrapper.find('p').at(2).text().trim()).toBe(mokHeroe.first_appearance);
        // } else {
        //     expect(wrapper.find('p').at(1).text().trim()).toBe(mokHeroe.first_appearance);    
        // }

    });
    
});