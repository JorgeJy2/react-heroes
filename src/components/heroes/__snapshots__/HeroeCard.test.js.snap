// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in card should snaptshot 1`] = `
<div
  className="card ms-3"
  style={
    Object {
      "maxWidth": 540,
    }
  }
>
  <div
    className="row no-gutters"
  >
    <div
      className="col-md-4"
    >
      <img
        alt="Name"
        className="card-img"
        src="./assets/heroes/1.jpg"
      />
    </div>
    <div
      className="col-md-8"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
           
          Name
           
        </h5>
        <p
          className="card-text"
        >
           
          None
           
        </p>
        <p
          className="card-text"
        >
          fly
           
        </p>
        <p
          className="card-text"
        >
          <small
            className="text-muted"
          >
            Today
          </small>
        </p>
        <Link
          to="./heroe/1"
        >
          More
        </Link>
      </div>
    </div>
  </div>
</div>
`;
